server/src/main/java/com/cloud/api/query/QueryManagerImpl.java

```diff
@@ -617,7 +618,9 @@ private Pair<List<UserAccountJoinVO>, Integer> getUserListInternal(Account calle
             ssc.addOr("email", Op.LIKE, "%" + keyword + "%");
             ssc.addOr("state", Op.LIKE, "%" + keyword + "%");
             ssc.addOr("accountName", Op.LIKE, "%" + keyword + "%");
-            ssc.addOr("accountType", Op.LIKE, "%" + keyword + "%");
+            if (EnumUtils.isValidEnum(Account.Type.class, keyword.toString().toUpperCase())) {
+                ssc.addOr("accountType", Op.EQ, EnumUtils.getEnum(Account.Type.class, keyword.toString().toUpperCase()));
+            }
```

Comment:
can this be simpler using `Account.Type.valueof(keyword)`? also I think `keyword` does not need a `.toString()`call.

Reply:
@DaanHoogland 
is valueOf case sensitive ?

Reply:
no, but a `toUpperCase()` can be added, and still the `toString()` can be dropped. It will return null if no value is found. I.E.
```suggestion
            try {
                if (Account.Type.valueOf(keyword.toUpperCase())) {
                    ssc.addOr("accountType", Op.EQ, EnumUtils.getEnum(Account.Type.class, keyword.toUpperCase()));
                }
            } catch ([IllegalArgumentException e) {
                logger.trace("keyword does not cantain an Account.Type");
            }
```
on second thought not really simpler or more legible, unless we let the exception trickle up. ´) never mind.